gtsam/discrete/DiscreteKey.h

```diff
@@ -73,6 +73,12 @@ namespace gtsam {
       push_back(key);
       return *this;
     }
+
+    /// Helper method to append additional discrete keys.
+    DiscreteKeys& append(const DiscreteKeys& keys) {
```

Comment:
why? It's a vector, use vector methods.

Reply:
I believe I was getting issues with constness when using the + operator or vector methods. Let me re-examine this.

Reply:
For the += operator on vectors, I get this error (since it is an operator overload from Boost)

```

In file included from /Users/varunagrawal/borglab/gtsam/gtsam/hybrid/HybridFactorGraph.cpp:13:
In file included from /Users/varunagrawal/borglab/gtsam/gtsam/hybrid/DCGaussianMixtureFactor.h:24:
In file included from /Users/varunagrawal/borglab/gtsam/gtsam/hybrid/DCFactor.h:15:
In file included from /Users/varunagrawal/borglab/gtsam/gtsam/discrete/DecisionTreeFactor.h:23:
In file included from /Users/varunagrawal/borglab/gtsam/gtsam/discrete/AlgebraicDecisionTree.h:21:
In file included from /Users/varunagrawal/borglab/gtsam/gtsam/discrete/DecisionTree-inl.h:24:
In file included from /opt/homebrew/include/boost/assign/std/vector.hpp:18:
/opt/homebrew/include/boost/assign/list_inserter.hpp:91:16: error: no matching member function for call to 'push_back'
            c_.push_back(boost::forward<T>(r));
            ~~~^~~~~~~~~
/opt/homebrew/include/boost/assign/list_inserter.hpp:414:13: note: in instantiation of function template specialization 'boost::assign_detail::call_push_back<std::vector<std::pair<unsigned long long, unsigned long>>>::operator()<gtsam::DiscreteKeys &>' requested here
            insert_(boost::forward<T>(t));
            ^
/opt/homebrew/include/boost/assign/list_inserter.hpp:406:13: note: in instantiation of function template specialization 'boost::assign::list_inserter<boost::assign_detail::call_push_back<std::vector<std::pair<unsigned long long, unsigned long>>>, std::pair<unsigned long long, unsigned long>>::insert<gtsam::DiscreteKeys &>' requested here
            insert(arg_type(), boost::forward<Ts>(ts)...);
            ^
/opt/homebrew/include/boost/assign/std/vector.hpp:42:16: note: in instantiation of function template specialization 'boost::assign::list_inserter<boost::assign_detail::call_push_back<std::vector<std::pair<unsigned long long, unsigned long>>>, std::pair<unsigned long long, unsigned long>>::operator()<gtsam::DiscreteKeys &>' requested here
        return push_back( c )( std::forward<V2>(v) );
               ^
/Users/varunagrawal/borglab/gtsam/gtsam/hybrid/HybridFactorGraph.cpp:89:10: note: in instantiation of function template specialization 'boost::assign::operator+=<std::pair<unsigned long long, unsigned long>, std::allocator<std::pair<unsigned long long, unsigned long>>, gtsam::DiscreteKeys &>' requested here
  result += dcKeys;
         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/vector:713:36: note: candidate function not viable: no known conversion from 'gtsam::DiscreteKeys' to 'const std::__vector_base<std::pair<unsigned long long, unsigned long>, std::allocator<std::pair<unsigned long long, unsigned long>>>::value_type' (aka 'const std::pair<unsigned long long, unsigned long>') for 1st argument
    _LIBCPP_INLINE_VISIBILITY void push_back(const_reference __x);
                                   ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/vector:716:36: note: candidate function not viable: no known conversion from 'gtsam::DiscreteKeys' to 'std::vector<std::pair<unsigned long long, unsigned long>>::value_type' (aka 'std::pair<unsigned long long, unsigned long>') for 1st argument
    _LIBCPP_INLINE_VISIBILITY void push_back(value_type&& __x);
                                   ^
1 error generated.
```

@ProfFan any ideas what this might be?

Reply:
I think this is because you have different types on both sides

Reply:
Just use insert with begin and end? If that does not work then let's go with append, but maybe rename it to push_back in analogy with factor graph method. "append" is yet another word.